spec/dynatrace_helper_spec.rb

```diff
@@ -644,4 +644,78 @@ def mock_dict (os, symbolsfile = Dir.pwd + "/spec/testdata/android-mapping-test.
       end
     end
   end
+
+  describe ".symlink_lldb" do
+    it "should raise error when destination_path does not exist" do
+      expect {
+        Fastlane::Helper::DynatraceHelper.symlink_lldb(anything, nil, anything)
+      }.to raise_error(RuntimeError)
+
+      expect {
+        Fastlane::Helper::DynatraceHelper.symlink_lldb(anything, "something/that/does/not/exist", anything)
+      }.to raise_error(RuntimeError)
+      verify_no_symlink_exists("something/that/does/not/exist")
+    end
+
+    context "when destination_path exists" do
+      before do
+        @destination_path = Dir.mktmpdir("destination-test")
+      end
+
+      it "should successfully create the symlink when lldb_path exists" do
+        lldb_path = Dir.mktmpdir("lldb-test")
+
+        Fastlane::Helper::DynatraceHelper.symlink_lldb(lldb_path, @destination_path, anything)
+
+        verify_symlink_exists(lldb_path, @destination_path)
+      end
+
+      context "when lldb_path does not exist" do
+        it "should successfully create the symlink when auto_link is true" do
+          expected_active_lldb_path = Fastlane::Helper::DynatraceHelper.active_lldb_path("#{%x(xcrun xcode-select --print-path)}".chomp)
+
+          Fastlane::Helper::DynatraceHelper.symlink_lldb(nil, @destination_path, true)
+
+          verify_symlink_exists(expected_active_lldb_path, @destination_path)
+        end
+
+        it "should delete existing symlinks and raise error when auto_link is not true" do
```

Comment:
Not quite sure if I understand this one correctly: There if the DSS client folder exists, but contains no LLDB and auto link is false we are deleting existing symlinks but fail? Why would we fail, couldn't Xcode be installed correctly and the DSS client resolved it automatically?

Reply:
default value of auto link is `true` and it can only set to `false` by using the provided option explicitly.

>  Why would we fail, couldn't Xcode be installed correctly and the DSS client resolved it automatically?

in that regard, this would basically mean that you do not want to use auto linking (since you set it to false), but also expecting it to be linked automatically 😅 which should be a failing case IMHO

Reply:
If auto link is not enabled, nothing should happen and behave like it currently does. Auto-linking is just an additional convenience features to catch some cases, where LLDB isn't resolved automatically by the OS/executable.

Reply:
I refactored the logic, please have a look and let me know what you think